blockstore/splitstore/README.md

```diff
@@ -3,129 +3,111 @@
 
-All new writes are directed to the hotstore, while reads first hit the
-hotstore, with fallback to the coldstore.
+All new writes are directed to the hotstore, while reads first hit the hotstore with fallback to the coldstore.
 
-Once 5 finalities have ellapsed, and every finality henceforth, the
-blockstore _compacts_.  Compaction is the process of moving all
-unreachable objects within the last 4 finalities from the hotstore to
-the coldstore. If the system is configured with a discard coldstore,
-these objects are discarded. Note that chain headers, all the way to
-genesis, are considered reachable. Stateroots and messages are
-considered reachable only within the last 4 finalities, unless there
-is a live reference to them.
+Once five finalities have elapsed and every subsequent finality, the blockstore _compacts_. Compaction is the process of moving all unreachable objects within the last four finalities from the hotstore to the coldstore. These objects are discarded if the system is configured with a discard coldstore. Chain headers are considered reachable all the way to the genesis block. Stateroots and messages are considered reachable only within the last four finalities unless there is a live reference to them.
 
 ## Compaction
 
 Compaction works transactionally with the following algorithm:
-- We prepare a transaction, whereby all i/o referenced objects through the API are tracked.
-- We walk the chain and mark reachable objects, keeping 4 finalities of state roots and messages and all headers all the way to genesis.
-- Once the chain walk is complete, we begin full transaction protection with concurrent marking; we walk and mark all references created during the chain walk. On the same time, all I/O through the API concurrently marks objects as live references.
-- We collect cold objects by iterating through the hotstore and checking the mark set; if an object is not marked, then it is candidate for purge.
+
+- We prepare a transaction whereby all i/o referenced objects through the API are tracked.
+- We walk the chain and mark reachable objects, keeping four finalities of state roots and messages and all headers all the way to genesis.
+- Once the chain walk is complete, we begin full transaction protection with concurrent marking; we walk and mark all references created during the chain walk. At the same time, all I/O through the API concurrently marks objects as live references.
+- We collect cold objects by iterating through the hotstore and checking the mark set; if an object is not marked, then it is a candidate for purge.
 - When running with a coldstore, we next copy all cold objects to the coldstore.
-- At this point we are ready to begin purging:
-  - We sort cold objects heaviest first, so as to never delete the consituents of a DAG before the DAG itself (which would leave dangling references)
```

Comment:
Nice catch removing sorting.
The "We delete in small batches ..." line is still relevant and should be kept around